File: Programing_Basics_with_JS/forLoopMoreExercises/11.OddEvenPosition.js
function oddEvenPosition(input) {
    let index = 0;
    let numbersOfInputs = Number(input[index]);
    index++;

    let oddSum = 0;
    let oddMin = Number.MAX_SAFE_INTEGER;
    let oddMax = Number.MIN_SAFE_INTEGER;
    let evenSum = 0;
    let evenMin = Number.MAX_SAFE_INTEGER;
    let evenMax = Number.MIN_SAFE_INTEGER;

    for(let i = 1;  i<= numbersOfInputs; i++) {
        let number = Number(input[index]);
        index++;

        if(i % 2 === 1) {
            oddSum += number;
            if(oddMin > number) {
                oddMin = number;
            } 
            if (oddMax < number) {
                oddMax = number;
            }
        } else {
            evenSum += number;
            if(evenMin > number) {
                evenMin = number;
            } 
            if (evenMax < number) {
                evenMax = number;
            }
        }
    }

    console.log(`OddSum=${oddSum.toFixed(2)},`);

    if(oddMin === Number.MAX_SAFE_INTEGER) {
        console.log('OddMin=No,');
    } else {
        console.log(`OddMin=${oddMin.toFixed(2)},`);
    }
    
    if(oddMax === Number.MIN_SAFE_INTEGER) {
        console.log('OddMax=No,')
    } else {
        console.log(`OddMax=${oddMax.toFixed(2)},`);
    }
    
    console.log(`EvenSum=${evenSum.toFixed(2)},`);

    if(evenMin === Number.MAX_SAFE_INTEGER) {
        console.log('EvenMin=No,');
    } else {
        console.log(`EvenMin=${evenMin.toFixed(2)},`);
    }

    if(evenMax === Number.MIN_SAFE_INTEGER) {
        console.log('EvenMax=No');
    } else {
        console.log(`EvenMax=${evenMax.toFixed(2)}`);
    }
}

oddEvenPosition(['6', '2', '3', '5', '4', '2', '1'])
oddEvenPosition(['5', '3', '-2', '8', '11', '-3'])
oddEvenPosition(['2', '1.5', '-2.5'])
oddEvenPosition(['4', '1.5', '1.75', '1.5', '1.75'])
oddEvenPosition(['1', '1'])
oddEvenPosition(['1', '-5'])
oddEvenPosition(['0'])
oddEvenPosition(['3', '-1', '-2', '-3'])

File: Programming_Basics_with_Python/for_loop_more_exercises/11._odd_even_position.py
import sys

total_numbers = int(input())

odd_sum = 0
odd_max = -sys.maxsize
odd_min = sys.maxsize
even_sum = 0
even_max = -sys.maxsize
even_min = sys.maxsize

for n in range(1, total_numbers + 1):
    number = float(input())
    if n % 2 == 1:
        odd_sum += number
        if number > odd_max:
            odd_max = number
        if number < odd_min:
            odd_min = number

    elif n % 2 == 0:
        even_sum += number
        if number > even_max:
            even_max = number
        if number < even_min:
            even_min = number

print(f'OddSum={odd_sum:.2f},')
if odd_min == sys.maxsize:
    print('OddMin=No,')
else:
    print(f'OddMin={odd_min:.2f},')
if odd_max == -sys.maxsize:
    print('OddMax=No,')
else:
    print(f'OddMax={odd_max:.2f},')

print(f'EvenSum={even_sum:.2f},')
if even_min == sys.maxsize:
    print('EvenMin=No,')
else:
    print(f'EvenMin={even_min:.2f},')
if even_max == -sys.maxsize:
    print('EvenMax=No')
else:
    print(f'EvenMax={even_max:.2f}')
